### PR TITLE
chore(dw): refactor otherUsage and getWorkloadName

### PR DESCRIPTION
**Pull request by gitdallas**
As part of https://issues.redhat.com/browse/RHOAIENG-5293

## Description
addressing a couple PR nits, it was already mostly taken care of throughout some other PRs that went in (moving things to utility classes/etc)

## How Has This Been Tested?
ran tests, ran app, made sure it still works

## Test Impact
updated tests with new data structure

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by mturley — APPROVED** [review]
LGTM

File: frontend/src/api/prometheus/distributedWorkloads.ts [hunk 1]
```diff
@@ -49,12 +49,12 @@ export type WorkloadCurrentUsage = {
...
 
 export type WorkloadWithUsage = {
-  workload: WorkloadKind | 'other';
+  workload: WorkloadKind;
   usage: number | undefined;
 };
 export type TopWorkloadsByUsage = Record<
   keyof WorkloadCurrentUsage,
-  { totalUsage: number; topWorkloads: WorkloadWithUsage[] }
+  { totalUsage: number; topWorkloads: WorkloadWithUsage[]; otherUsage: number | undefined }
```
Comment (andrewballantyne) [inline frontend/src/api/prometheus/distributedWorkloads.ts:57]: I am wondering if we should be looking at making this a type 🤔 I assume this is passed around... it's a type that is at least updated twice in this file alone

File: frontend/src/pages/distributedWorkloads/global/projectMetrics/sections/TopResourceConsumingWorkloads.tsx [hunk 2]
```diff
@@ -2,73 +2,82 @@ import * as React from 'react';
...
 import { truncateString } from '~/utilities/string';
 import { bytesAsPreciseGiB, roundNumber } from '~/utilities/number';
 import { WorkloadWithUsage } from '~/api';
-import { WorkloadKind } from '~/k8sTypes';
 
 interface TopResourceConsumingWorkloadsChartProps {
   metricLabel: string;
   unitLabel: string;
-  data: { totalUsage: number; topWorkloads: WorkloadWithUsage[] };
+  data: { totalUsage: number; topWorkloads: WorkloadWithUsage[]; otherUsage: number | undefined };
```
Comment (andrewballantyne) [inline frontend/src/pages/distributedWorkloads/global/projectMetrics/sections/TopResourceConsumingWorkloads.tsx:20]: Definitely need a type.

File: frontend/src/concepts/distributedWorkloads/utils.tsx [hunk 3]
```diff
@@ -149,6 +149,9 @@ export const getStatusCounts = (workloads: WorkloadKind[]): WorkloadStatusCounts
   return statusCounts;
 };
 
+export const getWorkloadName = (workload: WorkloadKind | 'other'): string =>
+  workload === 'other' ? 'Other' : workload.metadata?.name || 'Unnamed';
```
Comment (andrewballantyne) [inline frontend/src/concepts/distributedWorkloads/utils.tsx:153]: Correct me if I am wrong here -- but you never pass `other` 🤔

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by gitdallas** [comment]
@andrewballantyne -resolved

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**